packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -17,23 +20,35 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
   ) {}
 
   configure (configuration: InternalConfiguration<BrowserConfiguration>) {
+    this.logger = configuration.logger
+
     if (configuration.autoInstrumentNetworkRequests) {
       this.configEndpoint = configuration.endpoint
       this.xhrTracker.onStart(this.trackRequest)
       this.fetchTracker.onStart(this.trackRequest)
+      this.networkRequestCallback = configuration.networkRequestCallback
     }
   }
 
   private trackRequest = (startContext: RequestStartContext): RequestEndCallback | undefined => {
     if (!this.shouldTrackRequest(startContext)) return
 
+    const networkRequestInfo = this.networkRequestCallback({ url: startContext.url, type: startContext.type })
+
+    if (!networkRequestInfo) return
+
     const span = this.spanFactory.startSpan(
       `[HTTP]/${startContext.method.toUpperCase()}`,
       { startTime: startContext.startTime, makeCurrentContext: false }
     )
 
+    if (typeof networkRequestInfo.url === 'string') {
+      span.setAttribute('http.url', networkRequestInfo.url)
+    } else {
+      this.logger.warn(`expected url to be a string, got ${typeof networkRequestInfo.url}, http.url attribute discarded.`)
```

Comment:
not sure about the wording here @tomlongridge

Reply:
Maybe some context would be useful - "expected url to be a string following network request callback"